sno/timestamps.py

```diff
@@ -19,11 +19,33 @@ def to_iso8601_tz(timedelta):
     return f"+{abs_delta}" if abs(timedelta) == timedelta else f"-{abs_delta}"
 
 
+def iso8601_utc_to_datetime(iso8601z):
+    """
+    Accepts a string like: 2020-03-26T09:10:11Z
+    Returns a datetime.datetime object with UTC timezone.
+    """
+    return datetime.fromisoformat(iso8601z.replace("Z", "+00:00"))
+
+
+def iso8601_tz_to_timedelta(iso8601_tz):
+    """
+    Accepts a string like "+05:00" or "-05:00" (ie five hours ahead or behind).
+    Returns a datetime.timedelta object.
+    """
+    hours, minutes = iso8601_tz[1:].split(':')
+    hours = int(hours)
+    minutes = int(minutes)
+    r = timedelta(hours=int(hours), minutes=int(minutes))
+    if iso8601_tz[0] == '-':
+        r = -r
+    return r
+
+
 def commit_time_to_text(iso8601z, iso_offset):
     """
     Given an isoformat time in UTC, and a isoformat timezone offset,
     returns the time in a human readable format, for that timezone.
     """
-    right_time = datetime.fromisoformat(iso8601z.replace("Z", "+00:00"))
+    right_time = iso8601_utc_to_datetime(iso8601z)
     right_tzinfo = datetime.fromisoformat(iso8601z.replace("Z", iso_offset))
```

Comment:
This line of code contains a duplicate implementation of the iso8601_tz_to_timedelta function you have added above.
Since you've added that line, you may as well delegate to it. So this function would now be like this:
```
dt = datetime.fromisoformat(iso8601z.replace("Z", "+00:00"))
tz = datetime.timezone(iso8601_tz_to_timedelta(iso8601_tz))
return dt.astimezone(tz).strftime("%c %z")

As you can see, the one-liner is shorter but harder to read than the function you have written. If you like the function you have written, fine, but if you want to leverage the standard library's timezone parsing abilities - at the cost of some clarity perhaps - you could do the following:

_ARBITRARY_ISOFORMAT = datetime.utcfromtimestamp(0).isoformat()
def iso8601_tz_to_timedelta(iso8601_tz):
  return datetime.fromisoformat(_ARBITRARY_ISOFORMAT + iso8601_tz).tzinfo.utcoffset(None)

Reply:
ok. I like mine :)

Reply:
OK in 0893652c8da871cf31352bba535e809b76892af8 I did the first half of what you said, replacing the oneliner in `commit_time_to_text` with a call to my `iso8601_tz_to_timedelta` function. The previous implementation of `commit_time_to_text` was a bit frightening since it used a datetime with a UTC timezone which was _not_ in UTC.

Re the second comment, I like my implementation of `iso8601_tz_to_timedelta` just as much as your stdlib one, I think I'll keep it